code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -120,17 +120,27 @@
 			cdel(src)
 		return
 
-proc/flame_radius(radius = 1, turf/T) //~Art updated fire.
+proc/flame_radius(radius = 1, turf/T, burn_intensity = 25, burn_duration = 25, burn_damage = 25, fire_stacks = 25, int_var = 0.5, dur_var = 0.5, colour = "red") //~Art updated fire.
 	if(!T || !isturf(T))
 		return
-	if(radius < 0)
-		radius = 0
-	if(radius > 5)
-		radius = 5
+	radius = CLAMP(radius, 1, 7) //Sterilize inputs
+	int_var = CLAMP(int_var, 0.1,0.5)
+	dur_var = CLAMP(int_var, 0.1,0.5)
 	for(var/obj/flamer_fire/F in range(radius,T)) // No stacking flames!
 		cdel(F)
-	new /obj/flamer_fire(T, 5 + rand(0,11), 15, null, radius)
-
+	new /obj/flamer_fire(T, rand(burn_intensity*(0.5-int_var), burn_intensity*(0.5-int_var)) + rand(burn_intensity*(0.5-int_var), burn_intensity*(0.5-int_var)), rand(burn_duration*(0.5-int_var), burn_duration*(0.5-int_var)) + rand(burn_duration*(0.5-int_var), burn_duration*(0.5-int_var)), colour, radius) //Gaussian.
+	for(var/mob/living/carbon/M in range(radius, T))
+		if(isXeno(M))
+			var/mob/living/carbon/Xenomorph/X = M
+			if(X.fire_immune)
+				continue
+		if(M.stat == DEAD)
+			continue
+		var/dist = get_dist(T,M)
+		M.adjustFireLoss(rand(burn_damage*(0.5-int_var),burn_damage*(0.5+int_var)) + rand(burn_damage*(0.5-int_var),burn_damage*(0.5+int_var))*min(1,(1-(dist-1)/radius)))//Gaussian
+		M.adjust_fire_stacks(rand(burn_damage*(0.5-int_var),burn_damage*(0.5-int_var)) + rand(burn_damage*(0.5-int_var),burn_damage*(0.5-int_var))*min(1,(1-(dist-1)/radius)))//Gaussian
```

Comment:
Why such a complicated function for `fire_stacks`?

Reply:
It's a basic gaussian function; it just looks complicated.